tests/unit/sagemaker/serve/utils/test_hardware_detector.py

```diff
@@ -0,0 +1,209 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from mock.mock import patch
+import pytest
+
+from sagemaker.serve.utils import hardware_detector
+
+REGION = "us-west-2"
+VALID_INSTANCE_TYPE = "ml.p5.48xlarge"
+INVALID_INSTANCE_TYPE = "fl.c5.57xxlarge"
+DESCRIBE_INSTANCE_TYPE_RESULT = {
```

Comment:
Doing this might make the tests brittle since responses change with time/release, can we use moto or look into using the ec2 stub...

So can we use the mock ec2 service for interaction?

Reply:
We'll probably need to install a third party lib for EC2 stub. We can keep the tests simple like https://github.com/aws/sagemaker-python-sdk/blob/08bcc2ae23db25bc7d2f0c9668b58ebb76da03ec/tests/unit/test_collection.py#L32

I'll filter down the return value to only relevant fields.

Reply:
Is there a guidance around adding 3rd party dependencies to the SDK? Is it okay to add or do we need the Python SDK team to provide approval?